### ISSUE TITLE
Use numpy array as input

### ISSUE THREAD

**Issue by kpagels** [simontomaskarlsson/CycleGAN-Keras#20]
Hi,

I can see that your code requires image files as input e. g. Png, jpg etc. 

Is it possible in an easy way to use an numpy array as input? I only have the data as numpy array [6874, 128,128,1] and not ad images. 

Thanks in regards!

**Comment by simontomaskarlsson** [comment]
Hi @kpagels,

I guess it should be possible, when loading the image data it is also stored in arrays. Try to modify the loading function to fetch the data from where you have it, or hijack the code on row 206-211 and load your data instead.

Regards,
Simon

**Comment by kpagels** [comment]
Hi. Thanks for your answer! 
I was actually just on my way saying that I motified the code and fixed it today. Thanks.

If its okay I have two questions. 

1) What is a multi scale discriminator? I dont recall they talked about this in the paper?

2) Why brother using training and test set when the main purpose is to make the best mappings? Does it make sense for me to split up into training and test set?

**Comment by simontomaskarlsson** [comment]
Hi, 

Nice to hear! It is okey, I always try to answer when I have time over.

1. The multi scale discriminator is an inspiration from the UNIT model where you let the discriminator consist of more than one network,. You could say that the discriminators consists of multiple discriminators with different enconding depths.

2.  I see your point and you could argue to use as much data for training as possible. I do however think it is a good idea to always use images that have not been used during training when assessing how good the trained model works.

**Comment by kpagels** [comment]
Thanks. :)
